sympy/combinatorics/coset_table.py

```diff
@@ -766,14 +769,152 @@ def conjugates(self, R):
+            # abort the further generation of cosets
+            raise ValueError("the coset enumeration has defined more than %s cosets. Try with a greater value max number of cosets "%C.coset_table_max_limit)
+        self.table.append([None]*len(self.A))
+        H = self.subgroup
+        self._grp = (list(free_group(', ' .join(["a_%d" % i for i in range(len(H))])))).pop(0)
+        self.P.append([None]*len(self.A))
+        beta = len_table
+        self.p.append(beta)
+        # Define alpha^x = beta, beta^(x**-1) = alpha
+        self.table[alpha][self.A_dict[x]] = beta
+        self.table[beta][self.A_dict_inv[x]] = alpha
+        # P[alpha][x] = epsilon, P[beta][x**-1] = epsilon
+        self.P[alpha][self.A_dict[x]] = self._grp.identity
+        self.P[beta][self.A_dict_inv[x]] = self._grp.identity
+
+    def modified_scan(self, alpha, w, y, fill=False):
+        '''
+        Input: α∈Ω, w∈A*, y∈(Y∪Y-1)* with τ(α)w=G φ(y)τ(α).
+        Modified scan_and_fill if fill = True.
+        '''
+        while True:
+            r = len(w)
+            f = alpha
+            i = 0
+            f_p = self._grp.identity
+            while i < r and self.table[f][self.A_dict[w[i]]] is not None:
+                f_p = f_p*self.P[f][self.A_dict[w[i]]]
+                f = self.table[f][self.A_dict[w[i]]]
+                i += 1
+            if i >= r:
+                if f != alpha:
+                    self.modified_coincidence(f, alpha, f_p**-1*y)
+                return
+            b = alpha
+            b_p = y
+            j = r - 1
+            while j >= i and self.table[b][self.A_dict_inv[w[j]]] is not None:
+                b_p = b_p*self.P[b][self.A_dict_inv[w[j]]]
+                b = self.table[b][self.A_dict_inv[w[j]]]
+                j -= 1
+            if j < i:
+                self.modified_coincidence(f, b, f_p**-1*b_p)
+            elif j == i:
+                # Dedcution
+                self.table[f][self.A_dict[w[i]]] = b
+                self.table[b][self.A_dict_inv[w[i]]] = f
+                self.P[f][self.A_dict[w[i]]] = f_p**-1*b_p
+                self.P[b][self.A_dict_inv[w[i]]] = b_p**-1*f_p
+            else:
+                if not fill:
+                    self.modified_define(f, w[i])
+
+            # For scan an not filling.
+            if not fill:
+                break
+
+    def modified_scan_and_fill(self, alpha, w, y):
+        self.modified_scan(alpha, w, y, fill=True)
+
+    def modified_rep(self, k, p_p):
+        '''
+        Input: C, κ∈Ω, , pP
+        '''
+        p = self.p
+        lambda_ = k
+        rho = p[lambda_]
+        # Array to trace back compression path.
+        s = p[:]
+        while rho != lambda_:
+            s[rho] = lambda_
+            lambda_ = rho
+            rho = p[lambda_]
+        rho = s[lamda]
+        while rho != k:
+            mu = rho
+            rho = s[mu]
+            p[rho] = lambda_
+            p_p[rho] = p_p[rho]*p_p[mu]
+        return lambda_
```

Comment:
How different is this from the regular `rep` method? You have to update `p_p` but other than that it looks almost exactly the same.Would it be worth just letting `rep` have a `modified` keyword?

Reply:
And this should be merged with `rep`.